### PR TITLE
Sol2Ink - An easy transition from Solidity to Ink!

### PR DESCRIPTION
**Pull request by coreggon11**
### Project Abstract

Sol2Ink will be an application that will take code of already existing Solidity file and transpile it into Rust and ink! code
The generated code will need a double-check by a real developer, to fix some flaws but the hard work of rewriting Solidity to ink! will be done
This will ease the process of already existing projects/teams into Dotsama and ink! smart contracts ecosystem
The transpiler aims to use the standards and PSPs of Dotsama ecosystem
Sol2Ink should come first as a CLI app and then as a Web application
A prototype that can transpile Solidity interfaces into ink! traits was already tested by our team

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying?

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [X] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [X] I have read and understood the [FAQs](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md), [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md) and [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md).
- [X] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [X] I have read and acknowledge the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [X] The software delivered for this grant will be released under an open-source license specified in the application.
- [X] The initial PR contains only one commit (squash and force-push if needed).
- [X] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

### How Did You Hear About our grants program?

- [X] Personal Recommendation
- [X] Substrate Builders Program
- [X] Other: Supercolony applied for two Grants before

### CODE REVIEW

**Review by laboon — COMMENTED** [review]
Looks interesting! Do you have a link to the prototype you mentioned?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the grant application. This sounds like an interesting project. A few comments from my side:

- It might make sense to initially only start with milestone 1, just to see if it works? 
- Are you aware of similar solidity transpiler for other ecosystems that you might be able to leverage? 
- Regarding the Sol2Ink CLI of milestone 1, did you on purpose specify rust as the output here and not specifically ink!? 
- Just for reference a link to the other grant application by your team: https://github.com/w3f/Grants-Program/pull/921

**Comment by coreggon11** [comment]
Hello! Let me respond to all of your comments.

> Looks interesting! Do you have a link to the prototype you mentioned?

The prototype we currently have is in a private repository, and it was tested on one of our commercial projects, it is just a small piece of what we aim to do, but we saw how even this prototype was useful for us, so we decided to make it a full version.

> * It might make sense to initially only start with milestone 1, just to see if it works?

That is a good point. We can make it that way.

> * Are you aware of similar solidity transpiler for other ecosystems that you might be able to leverage?

We are aware of [Sol2Ligo](https://github.com/madfish-solutions/sol2ligo) transpiler used for easier migration to Tezos. We can also leverage parsing of Solidity code in [Solang](https://github.com/hyperledger-labs/solang/tree/main/solang-parser) as mentioned in the proposal. We can mention Sol2Ligo to the application as well.

> * Regarding the Sol2Ink CLI of milestone 1, did you on purpose specify rust as the output here and not specifically ink!?

What we meant by that point was the output of a Rust (.rs) file, of course, it will be Rust and ink! code, we will update that in the application.

**Comment by laboon** [comment]
@coreggon11 

_The prototype we currently have is in a private repository, and it was tested on one of our commercial projects, it is just a small piece of what we aim to do, but we saw how even this prototype was useful for us, so we decided to make it a full version._

Is it possible to see this prototype or, if not, some input and output from it?

**Comment by takahser** [comment]
@coreggon11 I love the idea, that would certainly be a very powerful tool! Are there any limits to the symbols/opcodes you would be able to transcompile from Solidity into Ink! Rust code or would any valid Solidity code work as an input to your CLI?

**Review by semuelle — CHANGES_REQUESTED** [review]

**Review by hakan-w3f — APPROVED** [review]
@coreggon11 , transpiling Solidity to ink! would be a very valuable help for those looking into porting their codes from Solidity.

File: applications/sol2ink.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# W3F Grant Proposal
+
+- **Project Name:** Sol2Ink
+- **Team Name:** [SuperColony](https://github.com/Supercolony-net)
+- **Payment Address:** ERC 20 ADDRESS:
+0xE1B19cE32866cDE87F8f59C7C2C5f90E093A6942
```
[the remark this answers is not in the document]
Comment (TtomaS7) [reply applications/sol2ink.md:6]: We want to get payment in USDT.

[the remark this answers is not in the document]
Comment (coreggon11) [reply applications/sol2ink.md:6]: I will add that to the application

**Review by Noc2 — APPROVED** [review]

**Review by takahser — APPROVED** [review]
@coreggon11 thanks for answering the questions. I'm happy to go ahead with it. 👍

**Review by hakan-w3f — APPROVED** [review]
@coreggon11 thanks for the prototype. Re-approving. Cheers.

**Review by laboon — APPROVED** [review]

**Pull request merged**